hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/DroolsRulesEngineImpl.java

```diff
@@ -155,36 +157,40 @@ public void fire() {
         // execution of the rules.  So, if we find multiple Data instances for the same Id, defer all but
         // the oldest to a subsequent run. Note that pendingData is already sorted by (id ASC, timestamp ASC) so
         // the iterator will present Data with the same id together, and time-ordered.
-        log.debug("firing rules...");
+        int initialPendingData = pendingData.size();
+        int initialPendingEvents = pendingEvents.size();
         int fireCycle = 0;
+        long startFiring = System.currentTimeMillis();
         while (!pendingData.isEmpty() || !pendingEvents.isEmpty()) {
+            log.debugf("Firing rules... PendingData [%s] PendingEvents [%s]", initialPendingData, initialPendingEvents);
 
-            if (log.isDebugEnabled()) {
-                log.debug("Data found. Firing rules on [" + pendingData.size() + "] datums and " +
-                        "[" + pendingEvents.size() + "] events.");
-            }
-
-            TreeSet<Data> batchData = new TreeSet<Data>(pendingData);
+            TreeSet<Data> batchData = new TreeSet<>(pendingData);
             Data previousData = null;
 
             pendingData.clear();
 
+            long startFiltering = System.currentTimeMillis();
```

Comment:
If you don't mind, please change timing variables to 'startBatching' and 'batchingTime'.